tests/test_report_manager.py

```diff
@@ -433,6 +433,61 @@ def _test_instance_group_helper(self, model_config_dict, expected_output):
         output = self.report_manager._create_instance_group_phrase(model_config)
         self.assertEqual(output, expected_output)
 
+    def test_gpu_name_and_memory(self):
+        gpu_info = {
+            'gpu_uuid1': {
+                'name': 'fake_gpu_name',
+                'total_memory': 2**30 * 2  # 2GB
+            },
+            'gpu_uuid2': {
+                'name': 'fake_gpu_name',
+                'total_memory': 2**30 * 2  # 2GB
+            },
+            'gpu_uuid3': {
+                'name': 'fake_gpu_name_2',
+                'total_memory': 2**30 * 4  # 4GB
+            },
+            'gpu_uuid4': {
+                'name': 'fake_gpu_name_3',
+                'total_memory': 2**30 * 8  # 8GB
+            }
+        }
+
+        report_manager = ReportManager(mode=MagicMock(),
+                                       config=MagicMock(),
+                                       gpu_info=gpu_info,
+                                       result_manager=MagicMock())
+
+        avg_gpu_metrics = {
+            'gpu_uuid1': {
+                "gpu_used_memory": 6000,
+                "gpu_utilization": 60
+            },
+            'gpu_uuid2': {
+                "gpu_used_memory": 6000,
+                "gpu_utilization": 60
+            },
+            'gpu_uuid4': {
+                "gpu_used_memory": 6000,
+                "gpu_utilization": 60
+            },
+        }
+
+        measurement = construct_run_config_measurement(
+            model_name=MagicMock(),
+            model_config_names=MagicMock(),
+            model_specific_pa_params=MagicMock(),
+            gpu_metric_values=avg_gpu_metrics,
+            non_gpu_metric_values=MagicMock(),
+            metric_objectives=MagicMock(),
+            model_config_weights=MagicMock())
+
+        measurements = [measurement, measurement]
```

Comment:
What happens when the two measurements have different gpu ids?

Reply:
updated the test to have one measurement use GPUs 1 and 4, and the other use 1 and 2.